Introduce run-webkit-tests --site-isolation

achristensen07 opened the pull request:
#### a53b3ee987184b39859fd3014d52a0266c47f2ce
<pre>
Introduce run-webkit-tests --site-isolation
<a href="https://bugs.webkit.org/show_bug.cgi?id=267028">https://bugs.webkit.org/show_bug.cgi?id=267028</a>

Reviewed by Pascoe.

This makes a convenient way to run tests in a cross-origin iframe with and without
site isolation enabled.  Any output difference is an indication that something needs
to be fixed.  I introduce a new TestExpectations file to allow us to see progress
as we fix things.

* LayoutTests/platform/mac-site-isolation/TestExpectations: Added.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):

Canonical link: <a href="https://commits.webkit.org/272619@main">https://commits.webkit.org/272619@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9969b05488807fa4e25d8f89dd5e71b8bea79701

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32403 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11145 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34235 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/34962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/29329 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33258 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13498 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/8349 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/34962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32827 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/9403 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/28966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/34962 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitpy~~](https://ews-build.webkit.org/#/builders/6/builds/32637 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/8333 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/28897 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/36301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/29464 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/29326 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/34450 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/8458 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/6407 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/32313 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/10119 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/9080 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4187 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/9028 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

pascoej commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -409,6 +412,15 @@ def parse_args(args):
             options.experimental_feature = []
         options.experimental_feature.append('UseGPUProcessForDOMRenderingEnabled=0')
 
+    if options.site_isolation:
```
I think this should go in `_set_up_derived_options`, we can also check `if port.port_name == "mac"` similar to the GPU process + remote layer tree's custom expectations.

You also have access to `options.platform` there, so it could be generic.

pascoej commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -352,6 +352,9 @@ def parse_args(args):
             help=("Enable all GPU process related features, also set additional expectations and the result report flavor.")),
+        optparse.make_option(
+            "--site-isolation", action="store_true", default=False,
+            help=("Run the test in a cross origin iframe with and without site isolation enabled and compare the results.  Uses site-isolation test expectations")),
```
nit: double space before last sentence.

aproskuryakov commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -352,6 +352,9 @@ def parse_args(args):
             help=("Enable all GPU process related features, also set additional expectations and the result report flavor.")),
+        optparse.make_option(
+            "--site-isolation", action="store_true", default=False,
+            help=("Run the test in a cross origin iframe with and without site isolation enabled and compare the results. Uses site-isolation test expectations")),
```
nit: is this "Run **each** test in a cross origin iframe..."? How does this deal with reftests?

achristensen07 replied:
Changing to each.  I'm not 100% sure how it handles reftests, but I think it does the same as all the other tests: run in an iframe with site isolation off, get output, run in an iframe with site isolation on, get output, compare.

achristensen07 committed:
Introduce run-webkit-tests --site-isolation
https://bugs.webkit.org/show_bug.cgi?id=267028

Reviewed by Pascoe.

This makes a convenient way to run tests in a cross-origin iframe with and without
site isolation enabled.  Any output difference is an indication that something needs
to be fixed.  I introduce a new TestExpectations file to allow us to see progress
as we fix things.

* LayoutTests/platform/mac-site-isolation/TestExpectations: Added.
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(parse_args):

Canonical link: https://commits.webkit.org/272619@main
